### PR TITLE
pythonPackages.pyctr: init at 0.6.0

### PR DESCRIPTION
**Pull request by rileyinman**
###### Motivation for this change
This library is used in a few programs I want to package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pyctr/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  propagatedBuildInputs = [ pycryptodomex ];
+
+  doCheck = pythonAtLeast "3.6";
+
+  meta = with lib; {
+    description = "Python library to interact with Nintendo 3DS files";
+    license = licenses.mit;
+    homepage = "https://github.com/ihaveamac/pyctr";
+  };
```
Comment (fabaff) [inline pkgs/development/python-modules/pyctr/default.nix:21]: New packages need a maintainer.

File: pkgs/development/python-modules/pyctr/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.6.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-05lMcsIeJIHI3LwHQTjr4M+bn1FG+GQscuGq34XxjK8=";
+  };
+
+  propagatedBuildInputs = [ pycryptodomex ];
+
+  doCheck = pythonAtLeast "3.6";
```
Comment (fabaff) [inline pkgs/development/python-modules/pyctr/default.nix:15]:
```suggestion
    disabled = pythonOlder "3.7";
```

see https://github.com/ihaveamac/pyctr/blob/master/setup.py#L12

**Review by fabaff — COMMENTED** [review]
Please add `pythonImportsCheck` as additional safe-guard.

**Comment by rileyinman** [comment]
Thanks for the feedback, I've integrated the suggestions!

File: pkgs/development/python-modules/pyctr/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "pyctr";
+  version = "0.6.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-05lMcsIeJIHI3LwHQTjr4M+bn1FG+GQscuGq34XxjK8=";
+  };
+
+  propagatedBuildInputs = [ pycryptodomex ];
+  pythonImportsCheck = [ "Cryptodome" ];
```
[the remark this answers is not in the document]
Comment (rileyinman) [reply pkgs/development/python-modules/pyctr/default.nix:14]: I see, I misunderstood what the check was for. Fixed!

File: pkgs/development/python-modules/pyctr/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-05lMcsIeJIHI3LwHQTjr4M+bn1FG+GQscuGq34XxjK8=";
+  };
+
+  propagatedBuildInputs = [ pycryptodomex ];
+
+  pythonImportsCheck = [ "pyctr" ];
+
+  disabled = pythonOlder "3.7";
```
Comment (jonringer) [inline pkgs/development/python-modules/pyctr/default.nix:17]: to be more aligned with other expressions. `disabled` should follow version
```suggestion
  version = "0.6.0";
  disabled = pythonOlder "3.7";

  src = fetchPypi {
    inherit pname version;
    sha256 = "sha256-05lMcsIeJIHI3LwHQTjr4M+bn1FG+GQscuGq34XxjK8=";
  };

  propagatedBuildInputs = [ pycryptodomex ];

  pythonImportsCheck = [ "pyctr" ];
```

Comment (rileyinman) [reply pkgs/development/python-modules/pyctr/default.nix]: Got it, thanks!

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 161602 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python310Packages.pyctr</li>    <li>python39Packages.pyctr</li>  </ul></details>

**Comment by rileyinman** [comment]
Thanks!

**Pull request merged**